src/main/java/org/folio/consortia/service/impl/SharingRoleCapabilitySetService.java

```diff
@@ -0,0 +1,138 @@
+    }
+    if (!sharingRoleRepository.existsByRoleId(roleId)) {
+      throw new ResourceNotFoundException("roleId", String.valueOf(roleId));
+    }
+  }
+
+  @Override
+  protected Set<String> findTenantsForConfig(SharingRoleCapabilitySetRequest request) {
+    return sharingRoleRepository.findTenantsByRoleIdAndSharedCapabilitySets(request.getRoleId());
+  }
+
+  @Override
+  protected void saveSharingConfig(List<SharingRoleEntity> sharingRoleEntityList) {
+    sharingRoleEntityList.forEach(sharingRoleEntity -> sharingRoleEntity.setIsCapabilitySetsShared(true));
+    sharingRoleRepository.saveAll(sharingRoleEntityList);
+  }
+
+  @Override
+  protected void deleteSharingConfig(UUID roleId) {
+    var sharingRoleEntityList = sharingRoleRepository.findByRoleId(roleId);
+    sharingRoleEntityList.forEach(sharingRoleEntity -> sharingRoleEntity.setIsCapabilitySetsShared(false));
+    sharingRoleRepository.saveAll(sharingRoleEntityList);
+  }
+
+  @Override
+  protected PublicationRequest createPublicationRequest(SharingRoleCapabilitySetRequest request,
+                                                        String httpMethod) {
+    PublicationRequest publicationRequest = new PublicationRequest();
+    publicationRequest.setMethod(httpMethod);
+    String url = request.getUrl();
+    UUID id = request.getRoleId();
+    if (httpMethod.equals(HttpMethod.PUT.toString()) || httpMethod.equals(HttpMethod.DELETE.toString())) {
+      url = url.replace("capability-sets", id + "/capability-sets");
+    }
+    publicationRequest.setUrl(url);
+    publicationRequest.setPayload(getPayload(request));
+    publicationRequest.setTenants(new HashSet<>());
+    return publicationRequest;
+  }
+
+  @Override
+  protected SharingRoleEntity createSharingConfigEntityFromRequest(SharingRoleCapabilitySetRequest request,
+                                                                   String tenantId) {
+    return sharingRoleRepository.findByRoleIdAndTenantId(request.getRoleId(), tenantId);
+  }
+
+  @Override
+  protected SharingRoleCapabilitySetResponse createSharingConfigResponse(UUID createRoleCapabilitySetsPCId,
+                                                                         UUID updateRoleCapabilitySetsPCId) {
+    return new SharingRoleCapabilitySetResponse()
+      .createRoleCapabilitySetsPCId(createRoleCapabilitySetsPCId)
+      .updateRoleCapabilitySetsPCId(updateRoleCapabilitySetsPCId);
+  }
+
+  @Override
+  protected SharingRoleCapabilitySetDeleteResponse createSharingConfigResponse(UUID publishRequestId) {
+    return new SharingRoleCapabilitySetDeleteResponse()
+      .pcId(publishRequestId);
+  }
+
+  @Override
+  protected ObjectNode updatePayload(SharingRoleCapabilitySetRequest request,
+                                     String sourceValue) {
+    var payload = objectMapper.convertValue(getPayload(request), ObjectNode.class);
+    return payload.set(SOURCE, new TextNode(sourceValue));
```

Comment:
For Role entity we should set 'type' property, related PR: https://github.com/folio-org/mod-roles-keycloak/pull/130/files#diff-3b0b02d25ff662c736fae80aadcd86fa23ba3201624cab6b908d4cfc58eaffa1R52